### PR TITLE
clean up NO_BIG_INT build, removing WOLFSSL_SP_MATH macro and heapmath compile

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
# Description

Enable option for building in no math library. Can be used in configurations like the following:

```
./configure --disable-fastmath --disable-heapmath --disable-sp-math-all --enable-cryptonly --disable-ecc --disable-rsa --disable-dh --disable-asn 
```

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -536,6 +536,25 @@ fi
 
 
 # MATH LIBRARY SELECTION
+
+# no math library
+AC_ARG_ENABLE([nomath],
```
Comment (dgarske) [inline configure.ac:541]: Did you also consider `./configure --disable-sp-math-all --disable-fastmath --disable-heapmath  --enable-cryptonly --disable-rsa --disable-ecc --disable-dh --disable-asn && make`?
If you would like to keep `nomath` consider adding a test case for it.

File: configure.ac [hunk 2]
```diff
@@ -536,6 +536,25 @@ fi
...
+    [ ENABLED_NOMATH=$enableval ],
+    [ ENABLED_NOMATH="no" ]
+    )
+
+if test "$ENABLED_NOMATH" = "yes"
+then
+    DEF_SP_MATH="no"
+    DEF_FAST_MATH="no"
+    ENABLED_SP_MATH_DEFAULT=no
+    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_NO_MATH"
```
Comment (dgarske) [inline configure.ac:552]: Are you able to use `NO_BIG_INT` instead?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (JacobBarthelmeh) [reply configure.ac:541]: I agree and do not like adding in another --enable- option. Having instead the --disable-(sp-math-all | fastmath | heapmath). In the case of user_settings.h builds would it remain where it automatically has (sp-math-all) though?

Comment (JacobBarthelmeh) [reply configure.ac:541]: Reading this comment and getting back to the PR I think we should keep the WOLFSSL_NO_MATH macro and have configure.ac set it if all math libraries are disabled. That way empty user_settings.h will use sp-math-all (easy default setup) and there is still a way for configure and user_settings.h builds to turn off all math libraries without a new --enable- flag.

**Review by dgarske — APPROVED** [review]

**Pull request merged**